spec/datadog/tracing/integration_spec.rb

```diff
@@ -312,13 +338,6 @@ def agent_receives_span_step3(previous_success)
         c.tracing.priority_sampling = priority_sampling if priority_sampling
       end
 
-      # Capture trace segments as they are about to be serialized
-      allow_any_instance_of(Datadog::Transport::Traces::Transport)
-        .to receive(:send_traces).and_wrap_original do |function, traces|
-          trace_segments.concat(traces)
-          function.call(traces)
-        end
-
```

Comment:
This approach was used by Single Span Sampling testing, but it's incomplete: it captures trace segments before they are formatted. Unformatted spans can look quite different from formatted ones.

This is specific snippet is now obsolete and actually conflicting with the new `trace_segments` capture at the beginning of this file above:
```ruby
      # Capture trace segments as they are about to be serialized
      segments = trace_segments
      allow_any_instance_of(Datadog::Transport::TraceFormatter)
        .to receive(:format!).and_wrap_original do |original|
          segments << original.call
        end
```